src/impl/tpls/KokkosBlas1_scal_tpl_spec_decl.hpp

```diff
@@ -60,165 +60,70 @@ inline void scal_print_specialization() {
-      double,                                                                  \
-      Kokkos::View<const double*, LAYOUT, Kokkos::Device<ExecSpace, MEMSPACE>, \
+      SCALAR_TYPE,                                                             \
+      Kokkos::View<const SCALAR_TYPE*, LAYOUT,                                 \
+                   Kokkos::Device<ExecSpace, MEMSPACE>,                        \
                    Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
       1, true, ETI_SPEC_AVAIL> {                                               \
-    typedef Kokkos::View<double*, LAYOUT, Kokkos::Device<ExecSpace, MEMSPACE>, \
+    typedef Kokkos::View<SCALAR_TYPE*, LAYOUT,                                 \
+                         Kokkos::Device<ExecSpace, MEMSPACE>,                  \
                          Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
         RV;                                                                    \
-    typedef double AV;                                                         \
-    typedef Kokkos::View<const double*, LAYOUT,                                \
+    typedef SCALAR_TYPE AV;                                                    \
+    typedef Kokkos::View<const SCALAR_TYPE*, LAYOUT,                           \
                          Kokkos::Device<ExecSpace, MEMSPACE>,                  \
                          Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
         XV;                                                                    \
     typedef typename XV::size_type size_type;                                  \
                                                                                \
-    static void scal(const RV& R, const double& alpha, const XV& X) {          \
-      Kokkos::Profiling::pushRegion("KokkosBlas::scal[TPL_BLAS,double]");      \
+    static void scal(const RV& R, const AV& alpha, const XV& X) {              \
+      Kokkos::Profiling::pushRegion("KokkosBlas::scal[TPL_BLAS," #SCALAR_TYPE  \
+                                    "]");                                      \
       const size_type numElems = X.extent(0);                                  \
       if ((numElems < static_cast<size_type>(INT_MAX)) &&                      \
           (R.data() == X.data())) {                                            \
         scal_print_specialization<RV, AV, XV>();                               \
-        int N   = numElems;                                                    \
-        int one = 1;                                                           \
-        HostBlas<double>::scal(N, alpha, R.data(), one);                       \
+        int N                            = numElems;                           \
+        int one                          = 1;                                  \
+        const BASE_SCALAR_TYPE alpha_val = alpha;                              \
```

Comment:
BASE_SCALAR_TYPE -> AV?

Reply:
I want to cast ```AV``` to ```BASE_SCALAR_TYPE```. Ex: ```Kokkos::complex<double>``` to ```std::complex<double>```
I change it to
```const BASE_SCALAR_TYPE alpha_b = static_cast<BASE_SCALAR_TYPE>(alpha);```
in the new commit